src/containers/CreateTaskRun/CreateTaskRun.js

```diff
@@ -0,0 +1,694 @@
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Dropdown,
+  Form,
+  FormGroup,
+  InlineNotification,
+  Modal,
+  TextInput
+} from 'carbon-components-react';
+import { ALL_NAMESPACES, generateId } from '@tektoncd/dashboard-utils';
+import { KeyValueList } from '@tektoncd/dashboard-components';
+import { injectIntl } from 'react-intl';
+import {
+  ClusterTasksDropdown,
+  NamespacesDropdown,
+  PipelineResourcesDropdown,
+  ServiceAccountsDropdown,
+  TasksDropdown
+} from '..';
+import { getClusterTask, getTask } from '../../reducers';
+import { createTaskRun } from '../../api';
+import { getStore } from '../../store/index';
+
+import './CreateTaskRun.scss';
+
+const parseTaskInfo = (state, taskRef, namespace) => {
+  if (taskRef) {
+    const task =
+      state.kind === 'ClusterTask'
+        ? getClusterTask(state, taskRef)
+        : getTask(state, { name: taskRef, namespace });
+    const paramSpecs = task?.spec?.params;
+    const resourceSpecs = task?.spec?.resources;
+    const taskError = !task;
+    return { resourceSpecs, paramSpecs, taskError };
+  }
+  return {};
+};
+
+const initialTaskInfoState = () => {
+  return {
+    namespace: '',
+    paramSpecs: [],
+    resourceSpecs: [],
+    taskError: false,
+    taskRef: ''
+  };
+};
+
+const initialParamsState = paramSpecs => {
+  if (!paramSpecs) {
+    return {};
+  }
+  return paramSpecs.reduce(
+    (acc, param) => ({ ...acc, [param.name]: param.default || '' }),
+    {}
+  );
+};
+
+const initialResourcesState = resourceSpecs => {
+  if (!resourceSpecs) {
+    return {};
+  }
+  const resources = {
+    inputs: {},
+    outputs: {}
+  };
+  if (resourceSpecs.inputs) {
+    resources.inputs = resourceSpecs.inputs.reduce(
+      (acc, res) => ({ ...acc, [res.name]: '' }),
+      {}
+    );
+  }
+  if (resourceSpecs.outputs) {
+    resources.outputs = resourceSpecs.outputs.reduce(
+      (acc, res) => ({ ...acc, [res.name]: '' }),
+      {}
+    );
+  }
+  return resources;
+};
+
+// K8s label documentation comes from here:
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+const labelKeyRegex = new RegExp(
+  '^(([a-z0-9A-Z]([a-z0-9A-Z-.]*[a-z0-9A-Z])?){0,253}/)?([a-z0-9A-Z]([a-z0-9A-Z-_.]*[a-z0-9A-Z])?){1,63}$'
+);
+const labelValueRegex = new RegExp(
+  '^([a-z0-9A-Z]([a-z0-9A-Z-_.]*[a-z0-9A-Z])?){0,63}$'
+);
+const isValidLabel = (type, value) => {
+  const regex = type === 'key' ? labelKeyRegex : labelValueRegex;
+  return regex.test(value);
+};
```

Comment:
we should extract this to src/utils so it can be reused across the 2 create run pages and ensure they're consistent